iOS/Flows/Threads/ThreadCoordinator.swift

```diff
@@ -39,23 +36,14 @@ class ThreadCoordinator: PresentableCoordinator<Void> {
         super.start()
         
         self.threadVC.swipeInputDelegate.didTapAvatar = { [unowned self] in
-            self.presentProfilePicture()
+            self.presentProfile(for: User.current()!)
         }
     }
     
-    func presentProfilePicture() {
-        let vc = ModalPhotoViewController()
-        
-        // Because of how the People are presented, we need to properly reset the KeyboardManager.
-        vc.dismissHandlers.append { [unowned self] in
-            self.threadVC.becomeFirstResponder()
-        }
-        
-        vc.onDidComplete = { _ in
-            vc.dismiss(animated: true, completion: nil)
+    func presentProfile(for person: PersonType) {
+        let coordinator = ProfileCoordinator(with: person, router: self.router, deepLink: self.deepLink)
+        self.present(coordinator) { [unowned self] result in
+            self.finishFlow(with: result)
```

Comment:
Now we close the thread if a conversation was selected, and the list will scroll to the selected cell.